src/update-feeds/update-feeds.ts

```diff
@@ -124,8 +128,60 @@ export const runUpdateFeed = async (providerName: string, chain: Chain, chainId:
   });
 };
 
-// eslint-disable-next-line @typescript-eslint/require-await
-export const processBatch = async (batch: ReadDapiWithIndexResponse[]) => {
+export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponse[]) =>
+  batch
+    .map((dapiResponse: ReadDapiWithIndexResponse) => {
+      const signedData = getStoreDataPoint(dapiResponse.decodedDataFeed.dataFeedId);
+
+      return {
+        ...dapiResponse,
+        signedData,
+      };
+    })
+    .filter(({ signedData, updateParameters, dataFeedValue }) => {
+      if (!signedData) {
+        return false;
+      }
+
+      const offChainValue = ethers.BigNumber.from(signedData.encodedValue);
+      const offChainTimestamp = Number.parseInt(signedData?.timestamp ?? '0', 10);
+      const deviationThreshold = updateParameters.deviationThresholdInPercentage;
+
+      return checkUpdateConditions(
+        dataFeedValue.value,
+        dataFeedValue.timestamp,
+        offChainValue,
+        offChainTimestamp,
+        updateParameters.heartbeatInterval,
+        deviationThreshold
+      );
+    });
+
+export const updateFeeds = async (_batch: ReturnType<typeof getFeedsToUpdate>, _chainId: string) => {
+  // TODO implement
+  // batch, execute
+};
+
+export const processBatch = async (batch: ReadDapiWithIndexResponse[], chainId: string) => {
   logger.debug('Processing batch of active dAPIs', { batch });
-  // TODO: Implement.
+
+  updateState((draft) => {
+    for (const dapi of batch) {
+      const receivedUrls = dapi.signedApiUrls.flatMap((url) =>
+        dapi.decodedDataFeed.beacons.flatMap((dataFeed) => `${url}/${dataFeed.airnodeAddress}`)
+      );
+
+      draft.signedApiUrlStore = receivedUrls.flat();
```

Comment:
These need to be indexed by provider. I think there is some confusion here.

We agreed on previous call was to store the latest URLs reported by each provider (these are to be replaced each time a batch is processed). When we call the signed APIs, we use all signed API URLs and for each beacon we save the latest signed data (based off the timestamp).

Reply:
Done. In retrospect an obvious requirement given that we replace URLs.